core/src/setup/Kokkos_Setup_Cuda.hpp

```diff
@@ -45,7 +45,18 @@
 #ifndef KOKKOS_CUDA_SETUP_HPP_
 #define KOKKOS_CUDA_SETUP_HPP_
 
-#if defined(KOKKOS_ENABLE_CUDA) && defined(__CUDACC__)
+#if !defined(KOKKOS_ENABLE_CUDA)
+#error \
+    "KOKKOS_ENABLE_CUDA was not defined, but Kokkos_Setup_Cuda.hpp was included anyway."
+#endif
+
+#if defined(KOKKOS_ENABLE_CUDA) && !defined(__CUDACC__)
+#error \
+    "KOKKOS_ENABLE_CUDA defined but the compiler is not defining the __CUDACC__ macro as expected"
+// Some tooling environments will still function better if we do this here.
+#define __CUDACC__
```

Comment:
Does that actually help right after an `#error` directive?! o_0

Reply:
Yup! CLion went from useless overload of hundreds of red squiglies in many of the files to literally only "what is this `<<<` stuff?"